### PR TITLE
Fixing tests and integrating to use the getRecordsCache.

### PR DESCRIPTION
**Pull request by sahilpalvia**

### CODE REVIEW

File: src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java [hunk 1]
```diff
@@ -1290,7 +1319,13 @@ public KinesisClientLibConfiguration withMaxCacheByteSize(final int maxCacheByte
     }
 
     public KinesisClientLibConfiguration withDataFetchingStrategy(String dataFetchingStrategy) {
-        recordsFetcherFactory.setDataFetchingStrategy(DataFetchingStrategy.valueOf(dataFetchingStrategy));
+        switch (dataFetchingStrategy.toUpperCase()) {
```
Comment (pfifer) [inline src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java]: You should be able to use DataFetchingStrategy.valueOf(dataFetchingStrategy) to get the enum value or throw an exception.

File: src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java [hunk 2]
```diff
@@ -1290,7 +1319,13 @@ public KinesisClientLibConfiguration withMaxCacheByteSize(final int maxCacheByte
...
+            case "PREFETCH_CACHED":
+                recordsFetcherFactory.setDataFetchingStrategy(DataFetchingStrategy.PREFETCH_CACHED);
+                break;
+            default:
+                recordsFetcherFactory.setDataFetchingStrategy(DataFetchingStrategy.DEFAULT);
```
Comment (pfifer) [inline src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java]: Should this be a default, or throw an exception?

File: src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisDataFetcher.java [hunk 3]
```diff
@@ -182,6 +188,19 @@ private String getIterator(Date timestamp) {
...
     }
+    
+    protected void sleepBeforeNextCall() {
+        if (lastResponseTime != null) {
+            long timeDiff = Duration.between(lastResponseTime, Instant.now()).abs().toMillis();
+            if (timeDiff < idleMillisBetweenCalls) {
+                try {
+                    Thread.sleep(idleMillisBetweenCalls - timeDiff);
+                } catch (InterruptedException e) {
+                    LOG.info("Thread interrupted, shutdown possibly called.");
```
Comment (pfifer) [inline src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisDataFetcher.java]: Is Thread.interrupted() checked by the callers?

File: src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/SimpleRecordsFetcherFactory.java [hunk 4]
```diff
@@ -21,8 +21,8 @@
 @CommonsLog
 public class SimpleRecordsFetcherFactory implements RecordsFetcherFactory {
```
Comment (pfifer) [inline src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/SimpleRecordsFetcherFactory.java]: Even though this is unchanged: 
Would it make more sense to have a records fetcher factory for each fetching strategy?  The Blocking strategy doesn't care about any of the configuration.

File: src/test/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/WorkerTest.java [hunk 5]
```diff
@@ -292,10 +293,22 @@ public void testWorkerLoopWithCheckpoint() {
         when(leaseCoordinator.getCurrentAssignments()).thenReturn(initialState).thenReturn(firstCheckpoint)
                 .thenReturn(secondCheckpoint);
 
-        Worker worker = new Worker(stageName, streamletFactory, config, streamConfig, INITIAL_POSITION_LATEST,
-                parentShardPollIntervalMillis, shardSyncIntervalMillis, cleanupLeasesUponShardCompletion, checkpoint,
-                leaseCoordinator, execService, nullMetricsFactory, taskBackoffTimeMillis, failoverTimeMillis,
-                KinesisClientLibConfiguration.DEFAULT_SKIP_SHARD_SYNC_AT_STARTUP_IF_LEASES_EXIST, shardPrioritization);
+        Worker worker = new Worker(stageName,
```
Comment (pfifer) [inline src/test/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/WorkerTest.java]: Can you not reformat existing code, it just increases the review length.

File: src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java (hunk 2 above)
Comment (sahilpalvia) [reply src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java]: Using value of takes care of this. It throws an IllegalArgumentException for unknown values.

**Pull request merged**